GDJS/Runtime/runtimescene.ts

```diff
@@ -35,89 +28,74 @@ namespace gdjs {
+      this._runtimeGame = runtimeGame;
+      this._variables = new gdjs.VariablesContainer();
+      this._timeManager = new gdjs.TimeManager();
+      this._onceTriggers = new gdjs.OnceTriggers();
+      this._requestedChange = SceneChangeRequest.CONTINUE;
+      this._cachedGameResolutionWidth = runtimeGame
+        ? runtimeGame.getGameResolutionWidth()
+        : 0;
+      this._cachedGameResolutionHeight = runtimeGame
+        ? runtimeGame.getGameResolutionHeight()
+        : 0;
+
       this._renderer = new gdjs.RuntimeSceneRenderer(
         this,
-        // @ts-ignore
+        // @ts-ignore This is needed because of test. They should mock RuntimeGame instead.
         runtimeGame ? runtimeGame.getRenderer() : null
       );
-      this._variables = new gdjs.VariablesContainer();
-      this._runtimeGame = runtimeGame;
-      this._timeManager = new gdjs.TimeManager();
-      this._requestedChange = SceneChangeRequest.CONTINUE;
+      this._debuggerRenderer = new gdjs.DebuggerRenderer(this);
 
       // What to do after the frame is rendered.
 
-      // Black background by default.
-
-      //An array used to create a list of all instance when necessary ( see _constructListOfAllInstances )
-      this._onceTriggers = new gdjs.OnceTriggers();
-
-      //The instances removed from the scene and waiting to be sent to the cache.
-
       // The callback function to call when the profiler is stopped.
       this.onGameResolutionResized();
     }
 
-    /**
-     * Activate or deactivate the debug visualization for collisions and points.
-     */
-    enableDebugDraw(
-      enableDebugDraw: boolean,
-      showHiddenInstances: boolean,
-      showPointsNames: boolean,
-      showCustomPoints: boolean
-    ): void {
-      if (this._debugDrawEnabled && !enableDebugDraw) {
-        this.getRenderer().clearDebugDraw();
-      }
-
-      this._debugDrawEnabled = enableDebugDraw;
-      this._debugDrawShowHiddenInstances = showHiddenInstances;
-      this._debugDrawShowPointsNames = showPointsNames;
-      this._debugDrawShowCustomPoints = showCustomPoints;
-    }
-
     /**
      * Should be called when the canvas where the scene is rendered has been resized.
      * See gdjs.RuntimeGame.startGameLoop in particular.
      */
     onGameResolutionResized() {
+      const oldGameResolutionOriginX = this.getViewportOriginX();
+      const oldGameResolutionOriginY = this.getViewportOriginY();
+      this._cachedGameResolutionWidth = this._runtimeGame
+        ? this._runtimeGame.getGameResolutionWidth()
+        : 0;
+      this._cachedGameResolutionHeight = this._runtimeGame
+        ? this._runtimeGame.getGameResolutionHeight()
+        : 0;
       for (const name in this._layers.items) {
         if (this._layers.items.hasOwnProperty(name)) {
           /** @type gdjs.Layer */
           const theLayer: gdjs.Layer = this._layers.items[name];
-          theLayer.onGameResolutionResized();
+          theLayer.onGameResolutionResized(
```

Comment:
Nice, any reason for moving oldGameRsolutionOriginX/Y here? Just make more sense?

Reply:
It allows to generalize it to custom objects that have an "internal" dimension change.